### PR TITLE
UCP/PROTO: Pass accurate memory registration flags

### PR DESCRIPTION
**Pull request by yosefe**
# Why
Pass registration flags according to the operation we are going to perform

### CODE REVIEW

**Comment by yosefe** [comment]
@brminich can you pls take a look?

File: src/ucp/proto/proto_multi.inl [hunk 1]
```diff
@@ -146,19 +146,19 @@ ucp_proto_multi_bcopy_progress(ucp_request_t *req,
...
 {
     ucs_status_t status;
 
     if (!(req->flags & UCP_REQUEST_FLAG_PROTO_INITIALIZED)) {
-        status = ucp_proto_request_zcopy_init(req, mpriv->reg_md_map,
-                                              comp_func);
+        status = ucp_proto_request_zcopy_init(req, mpriv->reg_md_map, comp_func,
+                                              uct_mem_flags);
         if (status != UCS_OK) {
+            // TODO fallback to other protocol
```
Comment (brminich) [inline src/ucp/proto/proto_multi.inl:161]: plz replace c++ style comment with `/**/`

Comment (yosefe) [reply src/ucp/proto/proto_multi.inl:161]: leftover:(

**Comment by yosefe** [comment]
@brminich force-pushed due to small PR

**Review by brminich — APPROVED** [review]

**Pull request merged**